persist/fs/mmap_other.go

```diff
@@ -27,7 +27,22 @@ import (
 	"syscall"
 )
 
+// mmap mmaps a file
 func mmap(fd, offset, length int64, opts mmapOptions) ([]byte, error) {
+	// MAP_PRIVATE because we only want to ever mmap immutable things and we don't
+	// ever want to propagate writes back to the underlying file
+	return mmapBase(fd, offset, length, syscall.MAP_PRIVATE, opts)
+}
+
+// mmapAnon requests a private (non-shared) region of anonymous (not backed by a file) memory from the O.S
+func mmapAnon(length int64, opts mmapOptions) ([]byte, error) {
+	// offset is 0 because we're not indexing into a file
+	// fd is -1 and MAP_ANON because we're asking for an anonymous region of memory not tied to a file
+	// MAP_PRIVATE because we don't plan on sharing this region of memory with other processes
+	return mmapBase(-1, 0, length, syscall.MAP_ANON|syscall.MAP_PRIVATE, opts)
+}
+
+func mmapBase(fd, offset, length int64, flags int, opts mmapOptions) ([]byte, error) {
```

Comment:
Maybe call this `mmapFd`?  Seems less OO-ish and more Go-ish, he.

Reply:
Actually on second thought, it might make it better to specialize the other two.  i.e. `mmap(...)` -> `mmapFd(...)` and `mmapAnon(...)` -> `mmapBytes(...)` and then `mmapBase(...)` -> `mmap(...)`

Reply:
sounds good